src/app/devices/audit-log/audit-log.component.ts

```diff
@@ -38,13 +38,17 @@ export class AuditLogComponent implements OnInit, AfterViewInit {
   public startIndex = new BehaviorSubject<number>(1)
   @ViewChild(MatSort, { static: true }) sort!: MatSort
 
-  constructor (public snackBar: MatSnackBar,
+  private subscription: Subscription = new Subscription()
+
+  constructor (
+    public snackBar: MatSnackBar,
     public readonly activatedRoute: ActivatedRoute,
     public readonly router: Router,
-    private readonly devicesService: DevicesService) { }
+    private readonly devicesService: DevicesService
+  ) { }
 
   ngOnInit (): void {
-    this.activatedRoute.params.subscribe(params => {
+    this.subscription = this.activatedRoute.params.subscribe(params => {
```

Comment:
this is explicitly called out in the angular docs not to do as activatedRoutes are insulated from the Router itself. The Router destroys a routed component when it is no longer needed. This means all the component's members will also be destroyed, including the injected [ActivatedRoute] and the subscriptions to its Observable properties. 

Please revert